components/region_cache_memory_engine/src/background.rs

```diff
@@ -154,10 +170,10 @@ impl BackgroundRunner {
         Self { engine_core }
     }
 
-    fn ranges_for_gc(&self) -> Vec<CacheRange> {
+    fn ranges_for_gc(&self) -> BTreeSet<CacheRange> {
```

Comment:
Minor: Can it be a more general Iterator (e.g. Box<dyn Iterator> maybe), if the caller does not need to use BTreeSet's specific method. It seems to me the caller does not care the type at all. It just needs to iterate all ranges.

Reply:
or ``impl Iterator<Item = CacheRange> `` if possible.